__tests__/app.test.js

```diff
@@ -85,35 +85,68 @@ describe("GET /api/articles/:article_id", () => {
 })
 
 describe("GET /api/articles", () => {
-    test("responds with an array of article objects", () => {
-        return request(app)
-        .get("/api/articles")
-        .expect(200)
-        .then(({ body }) => {
-          const { articles } = body;
-          expect(Array.isArray(articles)).toBe(true);
-          articles.forEach((article) => {
-              expect(article).toHaveProperty('author');
-              expect(article).toHaveProperty('title'); 
-              expect(article).toHaveProperty('article_id');
-              expect(article).toHaveProperty('topic'); 
-              expect(article).toHaveProperty('created_at'); 
-              expect(article).toHaveProperty('votes');
-              expect(article).toHaveProperty('article_img_url');
-              expect(article).toHaveProperty('comment_count');          
-          });
-       });
+  test("responds with an array of article objects", () => {
+    return request(app)
+    .get("/api/articles")
+    .expect(200)
+    .then(({ body }) => {
+      const { articles } = body;
+      console.log(articles)
+      expect(Array.isArray(articles)).toBe(true);
+      expect(articles.length).toBe(13)
+      articles.forEach((article) => {
+        expect.objectContaining({
+          author: expect.any(String),
+          title: expect.any(String),
+          article_id: expect.any(Number),
+          topic: expect.any(String),
+          created_at: expect.any(Number),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+          comment_count: expect.any(Number)
+        })         
+      });
+    });
+  })
+    test("articles should be sorted by date in descending order", () => {
+      return request(app)
+      .get("/api/articles")
+      .then(({body}) =>{
+        const { articles } = body;
+        console.log(articles)
+        expect(articles).toBeSortedBy('created_at', {descending : true})
+      })
     })
-    test("articles should be sorted by date in ascending order", () => {
-        return request(app)
-        .get("/api/articles")
-        .then(({body}) =>{
-            const { articles } = body;
-          expect(articles).toBeSortedBy('created_at', {descending : true})
-        })
+  test('200 - responds with an array of article objects with valid topic query', () => {
+    return request(app)
+    .get('/api/articles?topic=mitch')
+    .expect(200)
+    .then(({ body }) => {
+      const { articles } = body;
+      expect(Array.isArray(articles)).toBe(true);
```

Comment:
No need to check for an array when we are later using a forEach which proves it must be an array :)